src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -6807,6 +6809,12 @@ QueryProcessingStage::Enum MergeTreeData::getQueryProcessingStage(
     const StorageSnapshotPtr & storage_snapshot,
     SelectQueryInfo & query_info) const
 {
+    if (query_context->getClientInfo().collaborate_with_initiator)
+        return QueryProcessingStage::Enum::FetchColumns;
+
+    if (query_context->getSettingsRef().allow_experimental_parallel_reading_from_replicas && !query_context->getClientInfo().collaborate_with_initiator)
```

Comment:
I think here we also need to check `to_stage >= WithMergeableState`.
`to_stage` here is a stage up to which we want to execute a query. In theory it can be `FetchColumns`